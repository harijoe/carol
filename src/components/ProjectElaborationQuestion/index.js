import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { theme, breakpoint } from 'utils/style'
import cloudinary from 'utils/cloudinary'

import Icon from 'components/Icon'

const Wrapper = styled.div`
  position: relative;
  display: flex;
  flex-wrap: nowrap;
  align-items: flex-start;
  padding-bottom: ${theme('spaces.s')};
  padding-top: ${theme('spaces.m')};
`

const StyledIcon = styled(Icon)`
  position: relative;
  z-index: 1;
  width: 3.6rem;
  height: 3.6rem;
`

const BubbleQuestion = styled.div`
  position: relative;
  margin: 0 0 0 ${theme('spaces.s')};
  padding: ${theme('spaces.s')} ${theme('spaces.m')};
  overflow-wrap: break-word;
  width: ${props => (props.image ? '70%' : 'initial')};
  max-width: 70%;
  background: ${theme('colors.grayscale.lighter')};
  border-radius: 0 1rem 1rem;
  color: ${theme('colors.black')};
  white-space: pre-wrap;
  animation: 0.3s bubble ease-in 0s;

  @keyframes bubble {
    0% {
      opacity: 0;
      transform: translateY(-50%);
    }

    80% {
      opacity: 1;
    }

    100% {
      transform: translateY(0);
    }
  }

  &::before,
  &::after {
    position: absolute;
    top: 0;
    left: -1.2rem;
    height: 1.7rem;
    width: 1.2rem;
    content: '';
  }

  &::before {
    background: ${theme('colors.grayscale.lighter')};
  }

  &::after {
    background: ${theme('colors.white')};
    border-top-right-radius: 6rem;
  }
`

const QuestionBottomImage = styled.div`
  margin: ${theme('spaces.s')} -${theme('spaces.m')} -${theme('spaces.s')};
  background: url("${props => cloudinary(props.image, 'c_fill,h_286,w_539,g_south')}") center;
  background-size: cover;
  border-radius: 0 0 1rem 1rem;
  height: 18rem;
  ${breakpoint('m')`
    height: 26rem;
  `};
`

BubbleQuestion.displayName = 'BubbleQuestion'

const ProjectElaborationQuestion = ({ children, imageUrl, ...props }) => {
  if (children == null) {
    return null
  }

  return (
    <Wrapper {...{ ...props }}>
      <StyledIcon alt="quotatis" icon="logo" />
      <BubbleQuestion image={imageUrl}>
        {children}
        {imageUrl && <QuestionBottomImage image={imageUrl} />}
      </BubbleQuestion>
    </Wrapper>
  )
}

ProjectElaborationQuestion.propTypes = {
  children: PropTypes.any,
  imageUrl: PropTypes.string,
}

export default ProjectElaborationQuestion
