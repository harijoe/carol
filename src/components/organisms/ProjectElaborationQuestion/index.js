import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { theme } from 'utils/style'

import { Icon } from 'components'

const Wrapper = styled.div`
  position: relative;
  display: flex;
  flex-wrap: nowrap;
  align-items: flex-start;
  padding-bottom: ${theme('spaces.s')};
  padding-top: ${theme('spaces.m')};
`

const StyledIcon = styled(Icon)`
  position: relative;
  z-index: 1;
  width: 3.6rem;
  height: 3.6rem;
`

const BubbleQuestion = styled.div`
  position: relative;
  margin: 0 0 0 ${theme('spaces.s')};
  padding: ${theme('spaces.s')} ${theme('spaces.m')};
  overflow-wrap: break-word;
  max-width: 70%;
  background: ${theme('colors.white')};
  border-radius: 0 1rem 1rem 1rem;
  color: ${theme('colors.black')};
  box-shadow: 1px 1px 2px 0 rgba(19, 19, 19, 0.15);
  white-space: pre;

  &::before, 
  &::after {
    position: absolute;
    top: 0;
    left: -1.2rem;
    height: 1.7rem;
    width: 1.2rem;
    content: '';
  }

  &::before {
    background: ${theme('colors.white')};
  }

  &::after {
    background: ${theme('colors.grayscale.lightest')};
    border-top-right-radius: 6rem;
  }
`

BubbleQuestion.displayName = 'BubbleQuestion'

const ProjectElaborationQuestion = ({ children, ...props }) => {
  if (children == null) {
    return null
  }

  return (
    <Wrapper {...{ ...props }}>
      <StyledIcon
        alt="quotatis"
        icon="logo"
      />
      <BubbleQuestion>{children}</BubbleQuestion>
    </Wrapper>
  )
}

ProjectElaborationQuestion.propTypes = {
  children: PropTypes.any,
}

export default ProjectElaborationQuestion
