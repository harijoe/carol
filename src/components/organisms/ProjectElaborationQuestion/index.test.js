import React from 'react'
import { shallow } from 'enzyme'

import ProjectElaborationQuestion from './'

const wrap = (props = {}) => shallow(<ProjectElaborationQuestion {...props} />)

it('renders Question', () => {
  const wrapper = wrap({ children: 'my question' })

  expect(wrapper.find('BubbleQuestion')).toHaveLength(1)
  expect(wrapper).toMatchSnapshot()
})

it('renders null', () => {
  expect(wrap().node).toBeNull()
})
