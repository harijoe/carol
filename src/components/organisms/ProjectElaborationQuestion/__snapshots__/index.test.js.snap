// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Question 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProjectElaborationQuestion
    imageUrl="url"
  >
    my question
  </ProjectElaborationQuestion>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <StyledIcon
          alt="quotatis"
          icon="logo"
        />,
        <BubbleQuestion
          image="url"
        >
          my question
          <QuestionBottomImage
            image="url"
          />
        </BubbleQuestion>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "alt": "quotatis",
          "icon": "logo",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "my question",
            <QuestionBottomImage
              image="url"
            />,
          ],
          "image": "url",
        },
        "ref": null,
        "rendered": Array [
          "my question",
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "image": "url",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <StyledIcon
            alt="quotatis"
            icon="logo"
          />,
          <BubbleQuestion
            image="url"
          >
            my question
            <QuestionBottomImage
              image="url"
            />
          </BubbleQuestion>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "alt": "quotatis",
            "icon": "logo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "my question",
              <QuestionBottomImage
                image="url"
              />,
            ],
            "image": "url",
          },
          "ref": null,
          "rendered": Array [
            "my question",
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "image": "url",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
